data_model/src/trigger/mod.rs

```diff
@@ -428,6 +428,8 @@ mod tests {
                         .map(|_| ())
                         .unwrap()
                 }
+                // `TriggerExecuted` event does not chain trigger executions at the moment
+                FilterBox::TriggerExecuted(_) => unreachable!(),
```

Comment:
I thought these `unreachable`s might be temporary i.e. `TriggerExecuted` event might chain trigger executions.
If not, we could separate `Event` enum to avoid `unreachable`

> I don't see `TriggerExecuted` event emitted anywhere

You are right. I only defined the data structure in this PR